core/embed/lib/fonts/fonts.c

```diff
@@ -129,32 +132,108 @@ int font_baseline(int font) {
   return 0;
 }
 
+const uint8_t utf8_mapping(uint16_t c_2bytes) {
+  switch (c_2bytes) {
+    case 0xC381:
+      return 127;  // Á
+    case 0xC48C:
+      return 128;  // Č
+    case 0xC48E:
+      return 129;  // Ď
+    case 0xC389:
+      return 130;  // É
+    case 0xC49A:
+      return 131;  // Ě
+    case 0xC38D:
+      return 132;  // Í
+    case 0xC587:
+      return 133;  // Ň
+    case 0xC393:
+      return 134;  // Ó
+    case 0xC598:
+      return 135;  // Ř
+    case 0xC5A0:
+      return 136;  // Š
+    case 0xC5A4:
+      return 137;  // Ť
+    case 0xC39A:
+      return 138;  // Ú
+    case 0xC5AE:
+      return 139;  // Ů
+    case 0xC39D:
+      return 140;  // Ý
+    case 0xC5BD:
+      return 141;  // Ž
+    case 0xC3A1:
+      return 142;  // á
+    case 0xC48D:
+      return 143;  // č
+    case 0xC48F:
+      return 144;  // ď
+    case 0xC3A9:
+      return 145;  // é
+    case 0xC49B:
+      return 146;  // ě
+    case 0xC3AD:
+      return 147;  // í
+    case 0xC588:
+      return 148;  // ň
+    case 0xC3B3:
+      return 149;  // ó
+    case 0xC599:
+      return 150;  // ř
+    case 0xC5A1:
+      return 151;  // š
+    case 0xC5A5:
+      return 152;  // ť
+    case 0xC3BA:
+      return 153;  // ú
+    case 0xC5AF:
+      return 154;  // ů
+    case 0xC3BD:
+      return 155;  // ý
+    case 0xC5BE:
+      return 156;  // ž
+    default:
+      return 0;  // nonprintable
+  }
+}
+
 const uint8_t *font_get_glyph(int font, uint8_t c) {
-  c = convert_char(c);
-  if (!c) return 0;
+  uint16_t c_2bytes = convert_char_utf8(c);
+  bool is_printable = c_2bytes != 0x7F;
+  if (!c_2bytes) return 0;
+
+  // found UTF8 character
+  if (c_2bytes > 0xFF) {
+    c_2bytes = utf8_mapping(c_2bytes);
+    if (!c_2bytes) {
+      is_printable = false;
+    }
+  }
 
-  // printable ASCII character
-  if (c >= ' ' && c < 0x7F) {
+  // printable character
+  if (is_printable && c_2bytes >= ' ') {
```

Comment:
This check is not enough - we should also check that c_2bytes <= MAX_FONT_GLYPH_INDEX. Currently MAX_FONT_GLYPH_INDEX == 156 (used to be 126). Otherwise we introduce read out of bounds below when accessing `FONT_xxx_DATA`.

Reply:
` && c_2bytes <= 156` added in [8ad8475](https://github.com/trezor/trezor-firmware/pull/3206/commits/8ad8475eea41f7b66cbcd2b95f969ed0677a65ec)

Good suggestion for the future, noted it as TODO